### PR TITLE
Fix #1027: Ensure SC can be build with Java 17

### LINKED ISSUE #1027: Ensure SC can be build with Java 17 and Maven 3.8.x
Building SC with Java 17 targeting Java 11 does not work due to incompatible Maven Plugins.

```bash
class com.google.googlejavaformat.java.JavaInput (in unnamed module @0x751a1a56) cannot access class com.sun.tools.javac.parser.Tokens$TokenKind (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.parser to unnamed module @0x751a1a56
```

Investigate, if we can easily fix it to build with Java 17.

In addition, http blocker in Maven hits for a WARC module dependency.

### PR DESCRIPTION
**Pull request by rzo1**
Thanks for contributing to StormCrawler, your efforts are appreciated!

Developer Certificate of Origin
===============================

By contributing to StormCrawler, you accept and agree to the following terms and conditions (the *Developer Certificate of Origin*) for your present and future contributions submitted to StormCrawler. 
Please refer to the *Developer Certificate of Origin* section in [`CONTRIBUTING.md`](CONTRIBUTING.md) for details.

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

- Enables Matrix Build for Java 11 + Java 17 via GitHub actions
- Updates Code Format to 4.2
- If building with Java 17 locally, it might be required to setup `.mvn/jvm.config`, see https://github.com/Cosium/git-code-format-maven-plugin#jdk-16-peculiarities
- Maven Blocker did not happen locally for me on this machine, so might not be an issue.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -3,24 +3,29 @@
 
 name: Java CI with Maven
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+on: [push, pull_request]
```
Comment (jnioche) [inline .github/workflows/maven.yml:6]: so here we run  it regardless of the branch, correct?

File: pom.xml [hunk 2]
```diff
@@ -119,7 +119,7 @@
 			<plugin>
 				<groupId>com.cosium.code</groupId>
 				<artifactId>git-code-format-maven-plugin</artifactId>
```
Comment (jnioche) [inline pom.xml:121]: Shouldn't https://github.com/Cosium/git-code-format-maven-plugin#breaking-changes-between-3x-and-4x be addressed?

File: .github/workflows/maven.yml (hunk 1 above)
Comment (rzo1) [reply .github/workflows/maven.yml:6]: Yes - might be neat for feature branches without PR to have it.

File: pom.xml (hunk 2 above)
Comment (rzo1) [reply pom.xml:121]: Good catch. Wondering why it didn't break locally. Perhaps default config and ignored props paired with insufficent logs. Will fix that and give you a ping once done.

**Pull request merged**